tuya.cpp

```diff
@@ -820,6 +873,51 @@ void DeRestPluginPrivate::handleTuyaClusterIndication(const deCONZ::ApsDataIndic
         }
 
     }
+    // Time sync command
+    //https://developer.tuya.com/en/docs/iot/device-development/embedded-software-development/mcu-development-access/zigbee-general-solution/tuya-zigbee-module-uart-communication-protocol
+    else if (zclFrame.commandId() == 0x24)
+    {
+        DBG_Printf(DBG_INFO, "Tuya debug 1 : Time sync Request" );
+        
+        QDataStream stream(zclFrame.payload());
+        stream.setByteOrder(QDataStream::LittleEndian);
+        
+        quint16 payloadSize;
+        
+        stream >> payloadSize; // Always 0 for device > gateway
+        //other data ore useless
+        if (payloadSize == 0)
+        {
+            
+            quint32 time_now = 0xFFFFFFFF;              // id 0x0000 Time
+            //qint8 time_status = 0x0D;                   // id 0x0001 TimeStatus Master|MasterZoneDst|Superseding
+            qint32 time_zone = 0xFFFFFFFF;              // id 0x0002 TimeZone
+            quint32 time_dst_start = 0xFFFFFFFF;        // id 0x0003 DstStart
+            quint32 time_dst_end = 0xFFFFFFFF;          // id 0x0004 DstEnd
+            qint32 time_dst_shift = 0xFFFFFFFF;         // id 0x0005 DstShift
+            quint32 time_std_time = 0xFFFFFFFF;         // id 0x0006 StandardTime
+            quint32 time_local_time = 0xFFFFFFFF;       // id 0x0007 LocalTime
+            //quint32 time_valid_until_time = 0xFFFFFFFF; // id 0x0009 ValidUntilTime
+
+            DeRestPluginPrivate::getTime(&time_now, &time_zone, &time_dst_start, &time_dst_end, &time_dst_shift, &time_std_time, &time_local_time);
+
+            QByteArray data;
+            // Add UTC time
+            data.append((qint8)((time_now >> 24) & 0xff));
+            data.append((qint8)((time_now >> 16) & 0xff));
+            data.append((qint8)((time_now >> 8) & 0xff));
+            data.append((qint8)(time_now & 0xff));
+            // Ad local time
+            data.append((qint8)((time_local_time >> 24) & 0xff));
+            data.append((qint8)((time_local_time >> 16) & 0xff));
+            data.append((qint8)((time_local_time >> 8) & 0xff));
+            data.append((qint8)(time_local_time & 0xff));
+
```

Comment:
* Too much bit magic, use *QDataStream* instead;
* `QByteArray data` over shadows `quint32 data` from above;
* Always use camelCase instead snake_case when naming things, the whole code base is camel case.

```
QByteArray payload;
QDataStream stream2(&payload, QIODevice::WriteOnly);
stream2.setByteOrder(QDataStream::LittleEndian);

stream << timeNow;
stream << timeLocalTime;

```